assets/eip-7701/README.md

```diff
@@ -0,0 +1,166 @@
+the `ORIGIN` (`0x32`, `tx.origin`) and `CALLER` (`0x33`, `msg.sender`)
+are both equal to the address that is determined by the transaction's ECDSA signature (`yParity`, `r`, `s`).
+
+When processing an EIP-7701 transaction, however, multiple execution frames will be created.
+The full list of possible frames and their corresponding role definitions is as follows:
+
+1. **Validation Phase**
+    * `sender` deployment frame (once per account) - `role_sender_deployment`
+    * `sender` validation frame (required) - `role_sender_validation`
+    * `paymaster` validation frame (optional) - `role_paymaster_validation`
+2. **Execution Phase**
+    * `sender` execution frame (required) - `role_sender_execution`
+    * `paymaster` post-operation frame (optional) - `role_paymaster_post_op`
+
+All execution frames in the **Validation Phase** must be completed successfully without reverting
+in order for the transaction to be considered valid for a given position in a block.
+
+In all top-level frames, the global variables have the following meaning:
+
+| Opcode Name | Solidity Equivalent | Value                                                                                                           |
+|-------------|---------------------|-----------------------------------------------------------------------------------------------------------------|
+| `CALLER`    | `msg.sender`        | The `AA_ENTRY_POINT` address                                                                                    |
+| `ORIGIN`    | `tx.origin`         | The transaction `sender` address                                                                                |
+| `CALLDATA*` | `msg.data`          | Empty for all call frames except for the sender execution frame, for which it is set to `sender_execution_data` |
+
+## Transaction execution context
+
+Note that some behaviours in the EVM depend on the transaction context. These behaviours include:
+
+1. Costs of the `SSTORE (0x55)` opcode per [EIP-2200](./eip-2200)
+2. Costs of accessing cold addresses and slots per [EIP-2929](./eip-2929)
+3. Values available within the transient storage per [EIP-1153](./eip-1153)
+4. Maximum amount of gas refund assigned after the execution per [EIP-3529](./eip-3529)
+
+These features are not affected by the separation of the transaction into multiple frames.
+Meaning, for example, that a value set with `TSTORE (0x5D)` in one frame will remain available in the next one.
+
+### Costs of accessing cold addresses for Sender, Paymaster, and Deployer
+
+The Sender address is pre-warmed as part of the `AA_BASE_GAS_COST`.
+
+When a non-zero address that is not equal to the `Sender` address, is provided for a `Paymaster` or a `Deployer` contract,
+an additional EIP-2929 `COLD_ACCOUNT_READ_COST` cost of 2600 gas is charged and the address is added to `accessed_addresses`.
+
+### Flow diagrams
+
+#### Simple AA Transaction flow
```

Comment:
I haven't checked the diagrams but since a lot in this README doesn't match the current EIP, we need to check all of them and make sure they're up to date.